syntax = "proto3";

package query;

option go_package = "./proto;query";

message RequestQuery {
    string query = 1;
}

message User {
  string dorm = 1;
  int32 room_number = 2;
  string name = 3;
  string surname = 4;
  string phone = 5;
  string instagram = 6;
  string facebook = 7;
}

message Product {
  string name = 1;
  string photo = 2;
  string currency = 3;
  double price = 4;
  string category = 5;
  string description = 6;
  string action_type = 7;
  string person_id = 8;
}

message FullPayload {
  User user = 1;
  Product product = 2;
}

message FullPayloadList {
  repeated FullPayload payloads = 1;
}

service Query {
    rpc GetProductsByQuery(RequestQuery) returns (FullPayloadList);
}